### PR TITLE
[IMP] web_editor: provide video lazy loading and enhance privacy

### PR DESCRIPTION
**Pull request by SergeBayet**
[IMP] web_editor: enhanced privacy updating video 

This commit updates the YouTube video URL to "youtube-nocookie" for
improved privacy, and includes the `dnt` (do not track) parameter in
Vimeo videos to align with Google's initiative to phase out third-party
cookies by Q3 2024.

Requires : 
- https://github.com/odoo/enterprise/pull/59438

task-3724858

---
[IMP] web_editor: provide lazy loading

This commit introduces a method for lazy loading user-uploaded videos by 
using the loading="lazy" attribute on the video iframe tag.

task-3724858

### CODE REVIEW

**Review by Guillaume-gdi — COMMENTED** [review]
Good PR. :+1: Thanks.
I'm not an expert about data privacy stuff but I quickly checked what is sent to youtube.
image
Ok they are not using cookies anymore but why they need to know my OS...
They also write in the local storage.
image
with a device ID...
But yes they are not using cookies anymore... Let's say they need those info and t's already better.

In your commit message, you should indicate the task id like that: `task-2924927`

File: addons/web_editor/tools.py [hunk 1]
```diff
@@ -89,10 +89,12 @@ def get_video_url_data(video_url, autoplay=False, loop=False, hide_controls=Fals
             params['fs'] = 0
         if hide_yt_logo:
             params['modestbranding'] = 1
-        yt_extra = platform_match[1] or ''
+        yt_extra = platform_match[1] or '-nocookie'
```
[the remark this answers is not in the document]
Comment (SergeBayet) [reply addons/web_editor/tools.py:92]: Yep, i was not sure about that, but you're right. We can refactor this.

**Comment by SergeBayet** [comment]
@Guillaume-gdi Here's an article about YouTube's violation of GPDR's law : https://axbom.com/embed-youtube-videos-without-cookies/ That's very interesting...

**Comment by qsm-odoo** [comment]
@robodoo delegate=bvr-odoo 
@bvr-odoo

**Comment by SergeBayet** [comment]
@bvr-odoo Test suite has failed. I've rebased this PR.

**Comment by qsm-odoo** [comment]
@robodoo delegate=sobo-odoo 
@sobo-odoo

File: addons/web_editor/static/src/components/media_dialog/video_selector.js [hunk 2]
```diff
@@ -256,9 +256,11 @@ export class VideoSelector extends Component {
             div.innerHTML = `
                 <div class="css_editable_mode_display"></div>
                 <div class="media_iframe_video_size" contenteditable="false"></div>
-                <iframe frameborder="0" contenteditable="false" allowfullscreen="allowfullscreen"></iframe>
+                <iframe frameborder="0" contenteditable="false" allowfullscreen="allowfullscreen" title="Video"></iframe>
```
Comment (sobo-odoo) [inline addons/web_editor/static/src/components/media_dialog/video_selector.js:259]: Question: why was this change needed ? Maybe add an explanation in the commit message too :slightly_smiling_face:

File: addons/web_editor/static/src/components/media_dialog/video_selector.js [hunk 3]
```diff
@@ -256,9 +256,11 @@ export class VideoSelector extends Component {
...
             `;
-            div.querySelector('iframe').src = video.src;
+            const iframeEl = div.querySelector("iframe");
+            iframeEl.src = video.src
+            iframeEl.loading = "lazy";
```
Comment (sobo-odoo) [inline addons/web_editor/static/src/components/media_dialog/video_selector.js:263]: Nitpick but you could directly add it as an attribute in the HTML of the iframe right above, no ? :thinking:

**Review by sobo-odoo — CHANGES_REQUESTED** [review]
Here is my review :slightly_smiling_face: I tested and nothing seems broken, except the part about all videos starting at the same time mentioned in the task
Note that I didn't observe the other issue :shrug: : 
> "when you play one it pauses others"

I have a few comments :wink:

File: addons/web_editor/static/src/components/media_dialog/video_selector.js (hunk 3 above)
Comment (SergeBayet) [reply addons/web_editor/static/src/components/media_dialog/video_selector.js:263]: indeed :)

File: addons/web_editor/static/src/components/media_dialog/video_selector.js [hunk 4]
```diff
@@ -256,9 +256,11 @@ export class VideoSelector extends Component {
             div.innerHTML = `
```
[the remark this answers is not in the document]
Comment (SergeBayet) [reply addons/web_editor/static/src/components/media_dialog/video_selector.js:256]: I've split the commit into two separate commits.

File: addons/web_editor/static/src/components/media_dialog/video_selector.js (hunk 2 above)
Comment (SergeBayet) [reply addons/web_editor/static/src/components/media_dialog/video_selector.js]: I removed this. That's not the topic of this PR. You're right.

File: addons/web_editor/static/src/components/media_dialog/video_selector.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (SergeBayet) [reply addons/web_editor/static/src/components/media_dialog/video_selector.js:256]: Duly noted :)

**Comment by SergeBayet** [comment]
@sobo-odoo ready to review. Thanks.

File: addons/web_editor/static/src/components/media_dialog/video_selector.js (hunk 2 above)
Comment (sobo-odoo) [reply addons/web_editor/static/src/components/media_dialog/video_selector.js]: you could have kept it, I was just saying to explain it briefly :wink: Like "this commit also adds a title attribute to the video iframe, for accessibility (?) ." or something like that :slightly_smiling_face:

Comment (SergeBayet) [reply addons/web_editor/static/src/components/media_dialog/video_selector.js]: That would be better with the title of the Video to enhance accessibility. I'll keep this in mind and maybe create a task for it. "Video" for the title of a video is a little bit light.

File: addons/web_editor/tests/test_tools.py [hunk 5]
```diff
@@ -87,20 +87,22 @@ def test_get_video_url_data(self):
         for key in ['vimeo', 'vimeo_player']:
```
Comment (sobo-odoo) [inline addons/web_editor/tests/test_tools.py:87]: Commit message:
- First commit:
  - the commit title should start with a verb such that it completes the sentence "When applied, this commit will ..." so here "enhanced" is not right :wink: 
  - You left the two task references, I think only one is needed now that the commits are separated :slightly_smiling_face: 
 
- Second commit:
  - same comment for the task references :+1:

File: addons/web_editor/static/src/components/media_dialog/video_selector.js [hunk 6]
```diff
@@ -256,7 +256,7 @@ export class VideoSelector extends Component {
...
                 <div class="css_editable_mode_display"></div>
                 <div class="media_iframe_video_size" contenteditable="false"></div>
-                <iframe frameborder="0" contenteditable="false" allowfullscreen="allowfullscreen"></iframe>
+                <iframe loading="lazy" frameborder="0" contenteditable="false" allowfullscreen="allowfullscreen"></iframe>
```
Comment (sobo-odoo) [inline addons/web_editor/static/src/components/media_dialog/video_selector.js:259]: Question: here it sets as lazy the video uploaded when replacing an image for example, but it does not do it for the iframe of a background video. Shouldn't the background videos be lazy too ? :slightly_smiling_face:

**Review by sobo-odoo — CHANGES_REQUESTED** [review]
A few more comments :wink:

File: addons/web_editor/static/src/components/media_dialog/video_selector.js (hunk 6 above)
Comment (SergeBayet) [reply addons/web_editor/static/src/components/media_dialog/video_selector.js:259]: Yep, I didn't think about the BG videos. It's done.

File: addons/website/static/src/xml/website.background.video.xml [hunk 7]
```diff
@@ -8,6 +8,7 @@
                 </div>
             </div>
             <iframe t-att-id="iframeID"
+                    loading="lazy"
```
Comment (sobo-odoo) [inline addons/website/static/src/xml/website.background.video.xml:11]: Now that you modified a website file, you should add the website module in the commit title :wink:

**Review by sobo-odoo — CHANGES_REQUESTED** [review]
A last change and then I will r+ :+1: 

Also, a nitpick but in the enterprise PR, since the test only concerns the privacy, just keep the corresponding task reference in the commit message (I would have let it go, but since you have a last change to make, you can make the change there too :wink: )

**Review by sobo-odoo — APPROVED** [review]
Alright let's r+ ! :slightly_smiling_face: 

@robodoo rebase-ff r+
